### PR TITLE
Add KeylimeTpmError support and change return type in tpm.rs

### PR DESCRIPTION
**Pull request by leonjia0112**
Add support for tpm custom error type. This is aim on solving #40 for changing return type.
Implementation details:
1. Implement KeylimeTpmError enum has two contribution
    - error generate from TPM command execution
    - error generate from rust part of TPM interface
2. Implement From trait for Error:
    - `serde_json::error::Error`
    - `std::num::ParseIntError`    
    - `std::string::FromUtf8Error`
    - `std::io::Error`
    - `std::time::SystemTimeError`    
3. Upate return type of `run()` function in a tuple and a custom error
    `Result<(String, String), TpmExecError>`    
4. Support return type of other function in tpm.rs with `KeylimeTpmError`, which has benefit in code efficient and early return efficient

### CODE REVIEW

**Comment by frozencemetery** [comment]
@mbestavros PTAL

File: src/tpm.rs [hunk 1]
```diff
@@ -199,67 +190,47 @@ Following are function from tpm_quote.py program
...
-    };
-
-    let aik_password = match get_tpm_metadata_content("aik_pw") {
-        Ok(c) => c,
-        Err(e) => {
-            println!("Failed to get tpm aik_pw with error {}.", e);
-            return None;
+        None => {
+            return Err(KeylimeTpmError::new_tpm_rust_error(
+                "Can't retieve temp file path.",
```
Comment (mbestavros) [inline src/tpm.rs]: `retieve` -> `retrieve`

File: src/tpm.rs [hunk 2]
```diff
@@ -34,116 +35,109 @@ Following are function from tpm_initialize.py program
...
-    }
+    buffer.flush()?;
     Ok(())
 }
 
 /*
- * Input: None
- * Return: boolean
+ * Return:
+ *     true for vtpm/flase otherwise
```
Comment (mbestavros) [inline src/tpm.rs]: `flase` -> `false`

File: src/tpm.rs [hunk 3]
```diff
@@ -34,116 +35,109 @@ Following are function from tpm_initialize.py program
...
+                format!("Key: {} is missing in tpmdata.json", key).as_str(),
+            ))
         },
-        None => emsg("Key doesn't exist", None::<String>),
-    }
+        |content| {
+            content.as_str().map_or_else(
+                || {
+                    Err(KeylimeTpmError::new_tpm_rust_error(
+                        "Failed to convert Valur to stirng.",
```
Comment (mbestavros) [inline src/tpm.rs]: `Valur` -> `Value`

**Comment by leonjia0112** [comment]
@mbestavros thanks for catching those typos. Those are handled.

But for the improvement, I wasn't very sure what you mean. Do you suggest putting additional information instead of just using the description function?

**Pull request merged**